worldedit-core/src/main/java/com/fastasyncworldedit/core/FaweCache.java

```diff
@@ -137,6 +138,32 @@ public V load(@Nonnull T key) {
         });
     }
 
+    /**
+     * Create a new cache aimed to act as a thread-cache that is safe to the main server thread. If the method is called away
+     * from the main thread, it will return a {@link Function} referencing the {@link LoadingCache} returned by
+     * {@link FaweCache#createCache(Supplier)}. If it is called from the main thread, it will return a {@link Function} that
+     * will always return the result of the given {@link Supplier}. It is designed to prevent issues caused by
+     * internally-mutable and resettable classes such as {@link com.fastasyncworldedit.core.extent.filter.block.CharFilterBlock}
+     * from causing issues when used in edits on the main thread.
+     *
+     * @param withInitial The supplier used to determine the initial value if a thread cache is created, else to provide a new
+     *                    instance of the class being cached if on the main thread.
+     * @return a {@link Function} referencing a cache, or the given {@link Supplier}
+     * @since TODO
+     */
+    @AnnotationHelper.ApiDescription(info = "Designed for specific internal use.")
+    public <V> Function<Long, V> createMainThreadSafeCache(Supplier<V> withInitial) {
+        return new com.google.common.base.Function<>() {
```

Comment:
Is there are reason for using a google Function here? Also, can't we use a `LongFunction` instead?

Reply:
I had meant for it to be a java.util function, but LongFunction would make more sense